src/viam/sdk/robot/client.hpp

```diff
@@ -23,13 +25,50 @@ namespace viam {
 namespace sdk {
 
 using grpc::Channel;
-using viam::common::v1::ResourceName;
-using viam::common::v1::Transform;
-using viam::robot::v1::FrameSystemConfig;
 using viam::robot::v1::RobotService;
-using viam::robot::v1::Status;
+using time_point = std::chrono::time_point<long long, std::chrono::nanoseconds>;
+
+struct discovery_query {
+    discovery_query(){};
+    viam::robot::v1::DiscoveryQuery to_proto() const;
+    static discovery_query from_proto(const viam::robot::v1::DiscoveryQuery& proto);
+
+    std::string subtype;
+    std::string model;
+    friend bool operator==(const discovery_query& lhs, const discovery_query& rhs);
+};
+
+struct discovery {
+    discovery(){};
+    viam::robot::v1::Discovery to_proto() const;
+    static discovery from_proto(const viam::robot::v1::Discovery& proto);
+
+    discovery_query query;
+    AttributeMap results;
+    friend bool operator==(const discovery& lhs, const discovery& rhs);
+};
+
+struct frameSystemConfig {
+    viam::robot::v1::FrameSystemConfig to_proto() const;
+    static frameSystemConfig from_proto(const viam::robot::v1::FrameSystemConfig& proto);
+    frameSystemConfig(WorldState::transform frame) : frame(std::move(frame)){};
+
+    WorldState::transform frame;
+    AttributeMap kinematics;
+    friend bool operator==(const discovery_query& lhs, const discovery_query& rhs);
+};
+
+struct status {
+    status(){};
+    viam::robot::v1::Status to_proto() const;
+    static status from_proto(const viam::robot::v1::Status& proto);
+
+    std::optional<Name> name;
+    std::optional<AttributeMap> status_map;
+    std::optional<time_point> last_reconfigured;
```

Comment:
This is giving errors, but when I do try to use `boost::optional`, I automatically switches to `std::optional` when I run `ninja all && ninja install`. Not entirely sure what to do here.

Reply:
`std::optional` is only part of C++17, and the driver targets C++14. Somewhat unfortunately, our abseil dependency can automatically induce the driver to build in C++17 mode, since certain versions of abseil require it. Notably, the version of abseil in homebrew requires C++17, so builds on macOS are non-canonical and it is easy to accidentally C++17.

So, you need to use `boost::optional` here. I don't totally understand what you mean by " it automatically switches to std::optional". Can you elaborate, or, better, link to a log showing what you are seeing?

Reply:
Meanwhile, this reminded me to file https://viam.atlassian.net/browse/RSDK-6561

Reply:
I don't know how to elaborate and I'm not sure where the logs are for this. But I do quite literally mean, running `ninja all && ninja install` changes `boost::optional` to `std::optional`.

```
    boost::optional<Name> name;
    boost::optional<AttributeMap> status_map;
    boost::optional<time_point> last_reconfigured;
```
then I run: `ninja all && ninja install`, and I look at the same code again and it is now:

...Never mind. Now that I tried it again to show you, it is no longer automatically switching. I'm not sure what was happening there.